client/src/components/overview/GalleryView.jsx

```diff
@@ -232,13 +260,28 @@ function Overview({ currentStyleId, setCurrentStyleId }) {
       </StyledOverviewGrid>
     )
   } else {
+    // this is done using create context and higher order  component paradigm to showcase how it is supposed to work
     return (
       <StyledOverviewGrid>
-        <Expanded photos={styles[styleIndex].photos} expanded={expanded} setView={setView} imgIndex = {imgIndex} setImgIndex={setImgIndex}/>
+        <ExpandedContext.Provider value={{ photos: styles[styleIndex].photos, expandedVal: [expanded, setView], imgIndexVal: [imgIndex, setImgIndex] }}>
+          <Expanded>
+            <StyledIconSection>
+              {
+                styles[styleIndex].photos.map((photo, index) => {
+                  return <StyledIcon onClick={() => {
+                    setImgIndex(index);
+                  }}
+                  value={index} key={index} style={{backgroundImage: `url(${photo.url})`}}></StyledIcon>
+                })
+              }
+            </StyledIconSection>
+          </Expanded>
+        </ExpandedContext.Provider>
+        {/* <Expanded photos={styles[styleIndex].photos} expanded={expanded} setView={setView} imgIndex = {imgIndex} setImgIndex={setImgIndex}/> */}
       </StyledOverviewGrid>
```

Comment:
This is the compound rendering area and won't make much sense until you watch the video. It works though.